### DISCUSSION TITLE
dynamic window size

### DISCUSSION THREAD

**Discussion by monzeromer-lab** [slint-ui/slint#6304] [Q&A]
Hi all,

I've been trying to make the window size dynamic for my app, I've used the state and a property value assigned to the width and height of the window but all didn't work as I'm expecting

here is my code:

```slint
import { Button, CheckBox } from "std-widgets.slint";

export component AppComponent inherits Window {
    in-out property <string> caller: "Mohammed Ali";
    property <bool> open: false;
    width: 50px;
    height: 50px;
    background: #1e1e1e;

    animate width {
        duration: 500ms;
        easing: ease-in-out;
    }
    animate height {
        duration: 500ms;
        easing: ease-in-out;
    }

    label := Text {
        text: "InActive";
        color: #fff;
    }

    Timer {
        interval: 1s;
        running: false;
        triggered() => {
            open = !open;
        }
    }

    states [
        active when open: {
            root.width: 90px;
            root.height: 200px;
            label.text: "Active";
        }
        inactive when !open: {
            root.width: 50px;
            root.height: 50px;
        }
    ]
}
```

[Screencast from 2024-09-25 07-12-58.webm](https://github.com/user-attachments/assets/f88c8fe0-96de-4cad-a4fb-be6af85c2ff2)

it's seems to be working but the window size is still the same.

**Comment by bjorn** [comment]
Since you're demonstrating the issue in the live preview, I'm wondering whether you checked whether it's a live-preview issue or whether it also affects the actual app?

**Reply by monzeromer-lab** [reply]
Ooh, right.
that's a good point.
I honestly didn't think about that.
I'll try to just build an app with it to confirm.
